src/collection.jl

```diff
@@ -95,3 +95,32 @@ function fitscollection(basedir::String;
     end
     return df
 end
+
+
+# generator for image arrays specified by data frames (i.e. path of file, hdu etc.)
+@resumable function array(df::DataFrame)
+    for row in eachrow(df)
+        fh = FITS(row.path)
+        try
+            @yield getdata(fh[row.hdu])
+        finally
+            close(fh)
```

Comment:
Why not
```suggestion
        FITS(row.path) do fh
            @yield getdata(fh[row.hdu])
```
?

Reply:
This one throws an error:
```
syntax: label "_STATE_1" referenced but not defined
```

Reply:
Looks like a bug in `ResumableFunctions.jl`: https://github.com/BenLauwens/ResumableFunctions.jl/issues/28